packages/adminrouter/extra/src/test-harness/modules/generic_test_code/common.py

```diff
@@ -97,6 +98,67 @@ def generic_response_headers_verify_test(
             header_is_absent(resp.headers.items(), name)
 
 
+def generic_upstream_cookies_verify_test(
+        ar,
+        headers,
+        path,
+        cookies_to_send=None,
+        assert_cookies_present=None,
+        assert_cookies_absent=None):
+    """Test if cookies that are passed to the upstream by AR are correct
+
+    Helper function meant to simplify writing multiple tests testing the
+    same thing for different endpoints.
+
+    Arguments:
+        ar: Admin Router object, an instance of runner.(ee|open).Nginx
+        headers (dict): headers dict that contains JWT and cookies. The auth
+            data it contains must be valid.
+        path (str): path for which request should be made
+        cookies_to_send (dic): dictionary containing all the cookies that should
+            be send in the request
+        assert_cookies_present (dict): cookies to test where key is the
+            asserted cookie name and value is expected value of the cookie
+        assert_cookies_absent (dict): cookies that *MUST NOT* be present in the
+            upstream request
+    """
+    url = ar.make_url_from_path(path)
+    resp = requests.get(url,
+                        cookies=cookies_to_send,
+                        allow_redirects=False,
+                        headers=headers)
+
+    assert resp.status_code == 200
+
+    req_data = resp.json()
+
+    # Let's make sure that we got not more than one 'Cookie' header:
+    # https://tools.ietf.org/html/rfc6265#section-5.4
+    cookie_headers = []
+    for header in req_data['headers']:
+        if header[0] == 'Cookie':
+            cookie_headers.append(header)
+    assert len(cookie_headers) <= 1
+
+    if len(cookie_headers) == 1:
+        jar = cookies.SimpleCookie()
+        # It is a list containing a single tuple, we need the second element of
```

Comment:
:question: 
Nice comment, consider mentioning the tuple fields: "a single tuple `(name, value)`" if that is in fact what the tuple fields are